waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/CloseableThriftHiveMetastoreIfaceClientFactory.java

```diff
@@ -38,21 +44,41 @@ public CloseableThriftHiveMetastoreIfaceClientFactory(MetaStoreClientFactory met
   public CloseableThriftHiveMetastoreIface newInstance(AbstractMetaStore metaStore) {
     Map<String, String> properties = new HashMap<>();
     String uris = MetaStoreUriNormaliser.normaliseMetaStoreUris(metaStore.getRemoteMetaStoreUris());
-    String name = metaStore.getName().toLowerCase();
     properties.put(ConfVars.METASTOREURIS.varname, uris);
+    HiveConfFactory confFactory = new HiveConfFactory(Collections.<String> emptyList(), properties);
+    String name = metaStore.getName().toLowerCase();
+
     if (metaStore.getConnectionType() == TUNNELED) {
       MetastoreTunnel metastoreTunnel = metaStore.getMetastoreTunnel();
-      properties.put(WaggleDanceHiveConfVars.SSH_LOCALHOST.varname, metastoreTunnel.getLocalhost());
-      properties.put(WaggleDanceHiveConfVars.SSH_PORT.varname, String.valueOf(metastoreTunnel.getPort()));
-      properties.put(WaggleDanceHiveConfVars.SSH_ROUTE.varname, metastoreTunnel.getRoute());
-      properties.put(WaggleDanceHiveConfVars.SSH_KNOWN_HOSTS.varname, metastoreTunnel.getKnownHosts());
-      properties.put(WaggleDanceHiveConfVars.SSH_PRIVATE_KEYS.varname, metastoreTunnel.getPrivateKeys());
-      properties.put(WaggleDanceHiveConfVars.SSH_SESSION_TIMEOUT.varname, String.valueOf(metastoreTunnel.getTimeout()));
-      properties
-          .put(WaggleDanceHiveConfVars.SSH_STRICT_HOST_KEY_CHECKING.varname,
-              metastoreTunnel.getStrictHostKeyChecking());
+
+      sshSettings = buildSshSettings(metastoreTunnel);
+      metaStoreClientFactory.setTunnelableFactory(new TunnelableFactory<>(sshSettings));
```

Comment:
These settters should go, `metaStoreClientFactory` should not be changed by this class, what if the factory instance is injected somewhere else we'll have very strange sideeffect.

Reply:
Ok, I'll try to pass it directly to TunnelingMetaStoreClientFactory and update the PR once I have something that works.